client/src/main/java/io/pravega/keycloak/client/KeycloakAuthzClient.java

```diff
@@ -72,28 +87,24 @@ public String getRPT() {
             // obtain an access token with which to make an authorization request
             AccessTokenResponse accResponse;
             try {
-                accResponse = client.obtainAccessToken();
-                LOG.debug("Obtained access token from Keycloak");
+                accResponse = Retry.withExpBackoff(httpInitialDelayMs, 2, httpMaxRetries)
+                        .retryWhen(isRetryable()).run(() -> client.obtainAccessToken());
             } catch (HttpResponseException e) {
-                LOG.error("Failed to obtain access token from Keycloak", e);
-                if (e.getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
-                    throw new KeycloakAuthenticationException(e);
-                }
-                throw e;
+                LOG.error("Failed to obtain access token from Keycloak");
+                throw new KeycloakAuthenticationException(e);
             }
+            LOG.info("Obtained access token from Keycloak");
 
             // obtain an RPT
             AuthorizationRequest request = new AuthorizationRequest();
             try {
-                token = client.authorization(accResponse.getToken()).authorize(request);
-                LOG.debug("Obtained RPT from Keycloak");
+                token = Retry.withExpBackoff(httpInitialDelayMs, 2, httpMaxRetries)
+                        .retryWhen(isRetryable()).run(() -> client.authorization(accResponse.getToken()).authorize(request));
             } catch (HttpResponseException e) {
-                LOG.error("Failed to obtain RPT from Keycloak", e);
-                if (e.getStatusCode() == HttpStatus.SC_BAD_REQUEST) {
-                    throw new KeycloakAuthorizationException(e);
-                }
-                throw e;
+                LOG.error("Failed to obtain RPT from Keycloak");
```

Comment:
This log statement can be removed, as a method `isRetryable()` it calls already does the logging.